### PR TITLE
Resolve `VSCODE_REQUIRED_COMPONENTS` in VS Code verification errors

### LINKED ISSUE #939: Visual Studio Components template directive doesn't get rendered
**Describe the bug**
I tried to run `briefcase build windows VisualStudio` on a new Windows 11 box (Python 3.10) without VS installed and the error message didn't tell me which components I need.

**To Reproduce**
1. Do a clean Windows 11 installation
2. Install Python 3.10 from the Microsoft store
3. Open a command prompt
4. Run some commands:
```
cd \path\to\your\code
python3 -m pip install --upgrade --user pip setuptools wheel
python3 -m venv .venv
.\.venv\Scripts\activate
pip install briefcase
briefcase create windows VisualStudio
briefcase build windows VisualStudio
```
5. Receive this error message:
```
(.venv) C:\Users\winst\code\briefcase-windows-VisualStudio-template\tests\apps\verify-pyside2>briefcase build windows VisualStudio

Visual Studio does not appear to be installed. Visual Studio 2022 Community
Edition can be obtained as a free download from:

    https://visualstudio.microsoft.com/vs/community/

When you install Visual Studio, ensure you install the following workloads
and additional components:
{VSCODE_REQUIRED_COMPONENTS}
If you have Visual Studio installed in a non-default location, you must
either put MSBuild.exe in your path, or define an MSBUILD environment
variable that points at the MSBuild.exe provided by your Visual Studio
installation.
```

**Expected behavior**
`{VSCODE_REQUIRED_COMPONENTS}` gets rendered into a list of components I should install.

**Screenshots**
image

**Environment:**
 - Operating System: Windows 11 Pro, Version 22H2, build 22623.870
 - Python version: 3.10.8
 - Software versions:
   - Briefcase: 0.3.11
   - `pip freeze`:
```
arrow==1.2.3
binaryornot==0.4.4
briefcase==0.3.11
certifi==2022.9.24
chardet==5.0.0
charset-normalizer==2.1.1
click==8.1.3
colorama==0.4.5
commonmark==0.9.1
cookiecutter==2.1.1
gitdb==4.0.9
GitPython==3.1.29
idna==3.4
Jinja2==3.1.2
jinja2-time==0.2.0
MarkupSafe==2.1.1
packaging==21.3
platformdirs==2.5.2
psutil==5.9.3
Pygments==2.13.0
pyparsing==3.0.9
python-dateutil==2.8.2
python-slugify==6.1.2
PyYAML==6.0
requests==2.28.1
rich==12.6.0
six==1.16.0
smmap==5.0.0
text-unidecode==1.3
tomli==2.0.1
tomli_w==1.0.0
urllib3==1.26.12
```

### PR DESCRIPTION
**Pull request by rmartin16**
Issues
- Fixes #939 

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: src/briefcase/integrations/visualstudio.py [hunk 1]
```diff
@@ -159,10 +159,10 @@ def verify(cls, tools):
                 )
                 if not msbuild_path.exists():
                     raise BriefcaseCommandError(
-                        """\
+                        f"""\
 Your Visual Studio installation does not appear to provide MSBuild.
 Ensure that Visual Studio following workloads and components installed:
-{VSCODE_REQUIRED_COMPONENTS}
+{cls.VSCODE_REQUIRED_COMPONENTS}
```
Comment (drewbrew) [inline src/briefcase/integrations/visualstudio.py:165]: Thanks!

**Comment by drewbrew** [comment]
> so....windows threw up a BSOD and the disk has seemingly put itself in to a read-only mode to prevent further damage.

oh no 🙀

**Comment by freakboy3742** [comment]
@rmartin16 Were you expecting anything in addition to this patch? This PR looks pretty much good to go, but you've got it marked as a draft - were you expecting that it would need something else?

**Comment by rmartin16** [comment]
@freakboy3742 I don't think so. I was just about to test it when the SSD failed. I nearly have Windows set up in a new VM though; so, I should be able to finally get the testing done shortly.

**Review by freakboy3742 — APPROVED** [review]
👍

**Pull request merged**